File: .github/coercion/coercion.py
import io, os, sys, nsv

mode, indir, outdir = sys.argv[1], sys.argv[2], sys.argv[3]
for name in sorted(n for n in os.listdir(indir) if n.endswith('.nsv')):
    with open(os.path.join(indir, name), newline='') as f:
        if mode == 'batch':
            out = nsv.dumps(nsv.loads(f.read()))
        else:
            buf = io.StringIO()
            nsv.Writer(buf).write_rows(nsv.Reader(f))
            out = buf.getvalue()
    with open(os.path.join(outdir, name), 'w', newline='') as f:
        f.write(out)

File: .github/coercion/coercion.js
const fs = require('fs'), path = require('path');
const n = require('/tmp/nsv-js/index.js');
const [mode, indir, outdir] = process.argv.slice(2);

(async () => {
  for (const f of fs.readdirSync(indir).filter(f => f.endsWith('.nsv')).sort()) {
    const text = fs.readFileSync(path.join(indir, f), 'utf8');
    let out;
    if (mode === 'batch') {
      out = n.stringify(n.parse(text));
    } else {
      const rows = await new n.Reader(text).readRows();
      let buf = '';
      const sink = {
        write(chunk, enc, cb) {
          buf += chunk;
          const done = typeof enc === 'function' ? enc : cb;
          if (done) done();
          return true;
        },
      };
      await new n.Writer(sink).writeRows(rows);
      out = buf;
    }
    fs.writeFileSync(path.join(outdir, f), out);
  }
})();
